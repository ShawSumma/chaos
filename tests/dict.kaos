dict a = {'a': 1, 'b': 2}
print a
dict b = {'a': 1, 'b': 2, 'c': 3}
print b

json c = {"a": 1, "b": 2}
print c
json d = {"a": 1, "b": 2, "c": 3}
print d

bool dict dict1 = {'a': true, 'b': false}
print dict1
number dict dict2 = {'a': 1, "b": 2, 'c': 63.3, 'd': 12321.1515}
print dict2
string json dict3 = {'a': 'A', 'b': "asdasdaqs", 'c': 'asdasd123123', 'd': "."}
print dict3
