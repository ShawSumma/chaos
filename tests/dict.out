{'a': 1, 'b': 2}
{'a': 1, 'b': 2, 'c': 3}
{'a': 1, 'b': 2}
{'a': 1, 'b': 2, 'c': 3}
{'a': true, 'b': false}
{'a': 1, 'b': 2, 'c': 63.3, 'd': 12321.2}
{'a': 'A', 'b': 'asdasdaqs', 'c': 'asdasd123123', 'd': '.'}
[start] =>
	{name: a, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 3, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: b, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 3, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: c, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: c, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 3, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: d, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 3, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: c, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: dict1, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 0, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 0, 2nd_type: 0, value_type: 0, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 0, 2nd_type: 0, value_type: 0, role: 0} =>
	{name: dict2, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 1, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: c, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: d, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: dict3, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 2, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
	{name: (null), 2nd_name: (null), key: c, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
	{name: (null), 2nd_name: (null), key: d, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
[end]
[start] =>
	{name: a, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 3, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: c, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 3, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: d, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 3, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: c, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: dict1, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 0, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 0, 2nd_type: 0, value_type: 0, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 0, 2nd_type: 0, value_type: 0, role: 0} =>
	{name: dict2, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 1, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: c, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: (null), 2nd_name: (null), key: d, scope: main, type: 1, 2nd_type: 0, value_type: 2, role: 0} =>
	{name: dict3, 2nd_name: (null), key: (null), scope: main, type: 5, 2nd_type: 2, value_type: 4, role: 0} =>
	{name: (null), 2nd_name: (null), key: a, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
	{name: (null), 2nd_name: (null), key: b, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
	{name: (null), 2nd_name: (null), key: c, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
	{name: (null), 2nd_name: (null), key: d, scope: main, type: 2, 2nd_type: 0, value_type: 3, role: 0} =>
[end]
1
2
{'a': 5, 'b': 2, 'c': 3}
32
{'a': 5, 'b': 2, 'c': 3}
{'a': 15, 'b': 2}
{'b': 2, 'c': 3}
